doc/HOWTO/documenting-macros.md

```diff
@@ -0,0 +1,54 @@
+Documenting Macros
+==================
+
+The `make doc-nits` command (or the `check-docs` action) might tell
+you something like:
+
+```text
+include/openssl/bio.h: macro BIO_set_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_set_dgram_dest(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_dest(3) undocumented
+```
+
+and you'll want to document this.
+
+So, create a new .pod file in doc/man3/FUNCTION.pod.
+
+You might be documenting several related functions in that file,
+but don't try to think about giving it a neutral name.
+Just use the name of the first function as the file name,
+so in this case I called it:
+
+```text
+doc/man3/BIO_set_dgram_origin.pod
+```
+
+if you do use an unrelated name (like BIO_dgram.pod) then you'll get
+a warning about that.
+
+Then, you'll notice that your new file is not in test/build.info.
+That file is created by running:
+
+```shell
+make generate_doc_buildinfo
+```
+
+this will update doc/build.info.
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+```text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+```
+
+If it is the case that your interface is meant to be public, then you need
+to edit the file `util/other.syms` to add the names of your `#define` functions.
```

Comment:
If something is in include/openssl it is public by definition. So this sentence is misleading. And yeah, the warning from make doc-nits is misleading too.

Internal interfaces are in the other include directories and they are documented in doc/internal.

Reply:
I can't really know any of this.  If you have changes to suggest please make them.

Reply:
@mcr your sentence is indeed misleading, but I believe that the main reason why you are a bit confused about @t8m's change request is because you only know half of the story: 

Any new public symbol (i.e. a symbol which appears in a public header file) needs to be documented. Most of the times, this public symbol will be a public API function, in rare cases it can be a function-like macro, as in your case. 

Adding public functions is much more common than adding public macros, which explains the why the latter are stored in a file called `util/other.syms`.  More interesting are the files  `util/libcrypto.num` and  `util/libssl.num`, this is where all the public functions are stored. You don't edit those files manually, there is a make target to update them for you.

Before long, I will post a short explanation about the differences between adding a new public function and new public macro, which will hopefully make things more clear to you and enable you to complete your howto.